torch/distributed/_tensor/api.py

```diff
@@ -382,7 +383,7 @@ def distribute_tensor(
     tensor: torch.Tensor,
     device_mesh: Optional[DeviceMesh] = None,
     placements: Optional[Sequence[Placement]] = None,
-) -> DTensor:
+) -> Union[DTensor, 'XLAShardedTensor']:
```

Comment:
cc @wanchaol this might be more accurate and safe than returning `torch.Tensor`.

Reply:
Sure, I'm not sure if we care too much about type annotation or linting, if not we can simply append `type:ignore` to the `xla_distribute_tensor` callsite to turn that off. I am wondering if we should also consider refactor `XLAShardedTensor` to inherit from `DTensor`, so that we can get a clean and uniform representation to end users, can be done in follow up PRs

Reply:
Let's see if turning off the type annotation for `xla_distribute_tensor`. One concern I have is that the user might be confused with the unexpected return value type. We should explore typing `XLAShardedTensor` as `DTensor` offline and in follow up PRs.